src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BatchExecution.java

```diff
@@ -0,0 +1,217 @@
+
+            String sPrepStmt1 = "select count(*) from ctstable2 where TYPE_ID=?";
+
+            pstmt1 = connection.prepareStatement(sPrepStmt1);
+
+            for (int n = 1; n <= 3; n++) {
+                pstmt1.setInt(1, n);
+                rs = pstmt1.executeQuery();
+                rs.next();
+                retValue[i++] = rs.getInt(1);
+            }
+
+            pstmt1.close();
+
+            for (int j = 0; j < updateCount.length; j++) {
+                System.out.println("" + updateCount[j] + ", " + retValue[j]);
+                if (updateCount[j] != retValue[j] && updateCount[j] != Statement.SUCCESS_NO_INFO) {
+                    fail("executeBatch does not execute the Batch of SQL statements, Call to executeBatch is Failed!");
+                }
+            }
+        }
+        catch (BatchUpdateException b) {
+            fail("BatchUpdateException :  Call to executeBatch is Failed!");
+        }
+        catch (SQLException sqle) {
+            fail("Call to executeBatch is Failed!");
+        }
+        catch (Exception e) {
+            fail("Call to executeBatch is Failed!");
+        }
+    }
+
+    private static void createTable() throws SQLException {
+        String sql1 = "create table ctstable1 (TYPE_ID int, TYPE_DESC varchar(32), primary key(TYPE_ID)) ";
+        String sql2 = "create table ctstable2 (KEY_ID int,  COF_NAME varchar(32),  PRICE float, TYPE_ID int, primary key(KEY_ID), foreign key(TYPE_ID) references ctstable1) ";
+        stmt.execute(sql1);
+        stmt.execute(sql2);
+
+        String sqlin2 = "insert into ctstable1 values (1,'COFFEE-Desc')";
+        stmt.execute(sqlin2);
+        sqlin2 = "insert into ctstable1 values (2,'COFFEE-Desc2')";
+        stmt.execute(sqlin2);
+        sqlin2 = "insert into ctstable1 values (3,'COFFEE-Desc3')";
+        stmt.execute(sqlin2);
+
+        String sqlin1 = "insert into ctstable2 values (9,'COFFEE-9',9.0, 1)";
+        stmt.execute(sqlin1);
+        sqlin1 = "insert into ctstable2 values (10,'COFFEE-10',10.0, 2)";
+        stmt.execute(sqlin1);
+        sqlin1 = "insert into ctstable2 values (11,'COFFEE-11',11.0, 3)";
+        stmt.execute(sqlin1);
+
+    }
+
+    @BeforeAll
+    public static void testSetup() throws TestAbortedException, Exception {
+        assumeTrue(13 <= new DBConnection(connectionString).getServerVersion(),
+                "Aborting test case as SQL Server version is not compatible with Always encrypted ");
+        connection = DriverManager.getConnection(connectionString + ";columnEncryptionSetting=Enabled;");
+        stmt = (SQLServerStatement) connection.createStatement();
+        dropTable();
+        createTable();
+    }
+
+    private static void dropTable() throws SQLException {
+        stmt.executeUpdate("if object_id('" + "ctstable2" + "','U') is not null" + " drop table " + "ctstable2");
```

Comment:
Please use `Utils.dropObjectIfExists(tableName,"IsTable",stmt)` function to drop table.